sugondat-chain/pallets/blobs/src/tests.rs

```diff
@@ -377,3 +381,101 @@ fn test_pre_dispatch_max_total_blobs_size_reached() {
         );
     });
 }
+
+#[test]
+fn test_no_commitment_to_storage() {
+    let mut ext = new_test_ext();
+
+    macro_rules! execute_and_commit {
+        ($action:expr) => {
+            ext.execute_with(|| {
+                $action;
+            });
+            ext.commit_all()
+                .expect("Impossible having open transactions");
+        };
+    }
+
+    macro_rules! assert_key {
+        ([present] $key:expr) => {
+            match ext.as_backend().storage($key) {
+                Ok(Some(_)) => (),
+                _ => panic!("There must be some storage changing"),
+            }
+        };
+
+        ([non_present] $key:expr) => {
+            match ext.as_backend().storage($key) {
+                Ok(None) => (),
+                _ => panic!("There must be nothing storage changing"),
+            }
+        };
+    }
+
+    execute_and_commit!(assert_ok!(Blobs::submit_blob(
+        RuntimeOrigin::signed(alice()),
+        0,
+        get_blob(1024)
+    )));
+
+    assert_key!([present] & TotalBlobSize::<Test>::hashed_key());
+    assert_key!([present] & TotalBlobs::<Test>::hashed_key());
+    assert_key!([present] & BlobList::<Test>::hashed_key());
+
+    execute_and_commit!(Blobs::on_finalize(System::block_number()));
+
+    assert_key!([non_present] & TotalBlobSize::<Test>::hashed_key());
+    assert_key!([non_present] & TotalBlobs::<Test>::hashed_key());
+    assert_key!([non_present] & BlobList::<Test>::hashed_key());
+}
+
+#[test]
+fn test_storage_proof() {
+    let mut ext = new_test_ext();
+
+    macro_rules! execute_submit_blob {
+        () => {
+            System::on_initialize(System::block_number());
+            Blobs::on_initialize(System::block_number());
+
+            assert_ok!(Blobs::submit_blob(
+                RuntimeOrigin::signed(alice()),
+                0,
+                get_blob(1024)
+            ));
+
+            Blobs::on_finalize(System::block_number());
+            System::on_finalize(System::block_number());
+        };
+    }
+
+    ext.execute_with(|| {
+        System::set_block_number(0);
+        execute_submit_blob!();
+        System::set_block_number(1);
+    });
+
+    let (_res, proof) = ext.execute_and_prove(|| {
+        execute_submit_blob!();
+    });
+
+    let memory_bd = proof.into_memory_db();
+    let root = ext.backend.root();
+    let trie =
+        sp_trie::TrieDBBuilder::<LayoutV1<sp_core::Blake2Hasher>>::new(&memory_bd, root).build();
+
+    macro_rules! assert_key {
```

Comment:
macro for 1 line substitutions is overkill, can you remove?

Reply:
For sure. Initially, I was also expecting the presence of events in the proof. However, when I removed the check, I didn't update all the code properly. I will do it now